dev/CommandBarFlyout/CommandBarFlyoutCommandBar.cpp

```diff
@@ -85,75 +141,187 @@ void CommandBarFlyoutCommandBar::OnApplyTemplate()
+                            int deltaIndex = 1;
+                            int loopCount = 0;
+
+                            if (args.Key() == winrt::VirtualKey::Up)
+                            {
+                                deltaIndex = -1;
+                                startIndex = endIndex - 1;
+                                endIndex = -1;
+                            }
+
+                            do
+                            {
+                                // Give keyboard focus to the previous or next secondary command if possible
+                                for (int index = startIndex; index != endIndex; index += deltaIndex)
+                                {
+                                    auto secondaryCommand = SecondaryCommands().GetAt(index);
+
+                                    if (auto secondaryCommandAsControl = secondaryCommand.try_as<winrt::Control>())
+                                    {
+                                        if (secondaryCommandAsControl.FocusState() != winrt::FocusState::Unfocused)
+                                        {
+                                            focusedControl = secondaryCommandAsControl;
+                                        }
+                                        else if (focusedControl && IsControlFocusable(secondaryCommandAsControl, false /*checkTabStop*/) &&
+                                                 focusedControl != secondaryCommandAsControl)
+                                        {
+                                            if (FocusControl(
+                                                    secondaryCommandAsControl /*newFocus*/,
+                                                    focusedControl /*oldFocus*/,
+                                                    winrt::FocusState::Keyboard /*focusState*/,
+                                                    true /*updateTabStop*/))
+                                            {
+                                                args.Handled(true);
+                                                return;
+                                            }
+                                        }
+                                    }
+                                }
+                                loopCount++; // Looping again is needed when the last command has focus
+                            }
+                            while (loopCount < 2 && focusedControl && deltaIndex == 1);
+                        }
+                        args.Handled(true);
+                        break;
+                    }
+                    }
+                }
+            });
+        }
+        else
+        {
+            // UIElement.PreviewKeyDown is not available before RS3. Skip the custom Down/Up arrows handling above.
+            secondaryItemsRoot.AddHandler(winrt::UIElement::KeyDownEvent(), m_keyDownHandler, true);
```

Comment:
So what is the experience on RS2? Why can't the RS3+ handler be used on RS2 too?

Reply:
This was my thought too.  Can't we just use KeyDown instead of PreviewKeyDown?  Or does that have some sort of bad effect?

Reply:
The secondaryItemsRoot's KeyDown event handler is not being called when using the Up/Down arrows even with handledEventsToo==True.

On RS2, once on the top secondary command, the Up arrow brings you back to the MoreButton, instead of staying on that top command.